src/EFCore.SqlServer/Query/Internal/Translators/SqlServerDateTimeMemberTranslator.cs

```diff
@@ -69,8 +69,19 @@ public class SqlServerDateTimeMemberTranslator(
                     returnType),
 
             nameof(DateTime.Now)
+                when declaringType == typeof(DateTime)
                 => sqlExpressionFactory.Function(
-                    declaringType == typeof(DateTime) ? "GETDATE" : "SYSDATETIMEOFFSET",
+                    "GETDATE",
+                    arguments: [],
+                    nullable: false,
+                    argumentsPropagateNullability: [],
+                    returnType,
+                    typeMappingSource.FindMapping(typeof(DateTime), "datetime")),
+
+            nameof(DateTime.Now)
```

Comment:
nit:
 
```suggestion
            nameof(DateTimeOffset.Now)
```

Same below for DateTimeOffset.UtcNow